controllers/reconcile_test.go

```diff
@@ -198,6 +204,12 @@ func TestOpenTelemetryCollectorReconciler_Reconcile(t *testing.T) {
 								"app.kubernetes.io/managed-by": "opentelemetry-operator",
 								"app.kubernetes.io/part-of":    "opentelemetry",
 							})
+
+							sa := &v1.ServiceAccount{}
```

Comment:
I think the test already covers what was changed in the PR.

Reply:
not entirely – my worry is that on updates we will no longer be able to add _new_ annotations to service accounts because we no longer take the desired SA annotations and put them on the existing SA. this is good for external systems that are adding annotations as they will not be overridden, but would result in an odd user experience for ourselves where users can not add new annotations to their existing SA via the otelcol CR.

Reply:
https://github.com/open-telemetry/opentelemetry-operator/blob/6c923e39f36ae03cc8e10ce44a6d26c8eb32f530/internal/manifests/mutate.go#L66 enforces that desired annotations take precedence. 

The single line this PR removes ensures that user-defined annotations are added as well.

Reply:
I have added an extra annotation to `deploymentExtraPorts` and it is propagated to the SA after the update